src/types/UiaTextRangeBase.cpp

```diff
@@ -310,14 +310,319 @@ IFACEMETHODIMP UiaTextRangeBase::ExpandToEnclosingUnit(_In_ TextUnit unit) noexc
+            return [](const TextAttribute& attr) {
+                return !attr.IsBold() && !attr.IsFaint();
+            };
+        }
+    }
+    case UIA_ForegroundColorAttributeId:
+    {
+        // Expected type: VT_I4
+        THROW_HR_IF(E_INVALIDARG, val.vt != VT_I4);
+
+        // The foreground color is stored as a COLORREF.
+        const COLORREF queryForegroundColor{ base::ClampedNumeric<COLORREF>(val.iVal) };
+        return [queryForegroundColor](const TextAttribute& attr) {
+            return attr.GetForeground().GetRGB() == queryForegroundColor;
+        };
+    }
+    case UIA_IsItalicAttributeId:
+    {
+        // Expected type: VT_I4
+        THROW_HR_IF(E_INVALIDARG, val.vt != VT_BOOL);
+
+        // The text is either italic or it isn't.
+        const auto queryIsItalic{ val.boolVal };
+        if (queryIsItalic)
+        {
+            return [](const TextAttribute& attr) {
+                return attr.IsItalic();
+            };
+        }
+        else
+        {
+            return [](const TextAttribute& attr) {
+                return !attr.IsItalic();
+            };
+        }
+    }
+    case UIA_StrikethroughColorAttributeId:
+    {
+        // Expected type: VT_I4
+        THROW_HR_IF(E_INVALIDARG, val.vt != VT_I4);
+
+        // The strikethrough color is stored as a COLORREF.
+        // However, the text buffer doesn't actually store the color.
+        // Instead, we just store whether or not the text is crossed out.
+        // So we'll assume that _any_ query here (except 0) is interpreted as "is the
+        // text crossed out?"
+
+        if (val.iVal == 0)
+        {
+            return [](const TextAttribute& attr) {
+                return !attr.IsCrossedOut();
+            };
+        }
+        else
+        {
+            return [](const TextAttribute& attr) {
+                return attr.IsCrossedOut();
+            };
+        }
+    }
+    case UIA_StrikethroughStyleAttributeId:
+    {
+        // Expected type: VT_I4
+        THROW_HR_IF(E_INVALIDARG, val.vt != VT_I4);
+
+        // The strikethrough style is stored as a TextDecorationLineStyle.
+        // However, The text buffer doesn't have different styles for being crossed out.
+        // Instead, we just store whether or not the text is crossed out.
+        // So we'll assume that _any_ query here is interpreted as "is the
+        // text crossed out?" UNLESS we are asked for TextDecorationLineStyle_None
+        const TextDecorationLineStyle queryStrikethroughStyle{ static_cast<TextDecorationLineStyle>(val.iVal) };
+        if (queryStrikethroughStyle == TextDecorationLineStyle_None)
+        {
+            return [](const TextAttribute& attr) {
+                return !attr.IsCrossedOut();
+            };
+        }
+        else
+        {
+            return [](const TextAttribute& attr) {
+                return attr.IsCrossedOut();
+            };
+        }
+    }
+    case UIA_UnderlineColorAttributeId:

```

Comment:
Check with some accessibility folks as to whether "is underlined?" is a valid way to answer "where is the red underline?"

Reply:
@codeofdusk and I did some research. So, some apps (like MS Word) let you have a different color from the text vs the underline. Since console _only_ ever uses the foreground color, it doesn't make sense for us to support this.